### PR TITLE
FE-1219 - Alerts table mute button state gets reverted on first click

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-1219](https://sparkpost.atlassian.net/browse/FE-1219)

### What Changed
- Refactors Alerts list page to use functional components, hooks
- Fixes styling problems with "Recent Incidents" Panels - mobile <-- This was actually a little bugged - not functionally, but from a presentation standpoint

### How To Test
- Go to `/alerts` and verify the page renders and works as expected in both Hibana and the OG theme
- Create a new alert and toggle the "Mute" switch - verify it handles state correctly <-- this part is the bug fix.

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: src/pages/templates/components/RecentActivity.module.scss [hunk 1]
```diff
@@ -7,7 +7,7 @@
   margin-bottom: spacing('large');
 
   @media (min-width: breakpoint('large')) {
-    grid-template-columns: 1fr 1fr 1fr 1fr; // This ensuers that each Panel is full width on wider screens
+    grid-template-columns: 1fr 1fr 1fr 1fr; // This ensures that each Panel is full width on wider screens
```
Comment (nicklemmon) [inline src/pages/templates/components/RecentActivity.module.scss:10]: Just saw this typo while referencing these styles.

File: src/pages/alerts/components/AlertCollection.js [hunk 2]
```diff
@@ -1,78 +1,136 @@
...
-      { label: 'Metric', sortKey: 'metric' },
-      { label: 'Last Triggered', sortKey: 'last_triggered_timestamp' },
-      { label: 'Mute', sortKey: 'muted' },
-      { label: <ScreenReaderOnly>Actions</ScreenReaderOnly> },
-    ];
+function reducer(state, action) {
+  switch (action.type) {
+    case 'OPEN_DELETE_MODAL':
+      return { ...state, alert: action.alert, isDeleteModalOpen: true };
+    case 'CLOSE_DELETE_MODAL':
```
Comment (shrirupa) [inline src/pages/alerts/components/AlertCollection.js:26]: we could use useModal hook here.

File: src/pages/alerts/ListPage.js [hunk 3]
```diff
@@ -1,179 +1,111 @@
-import React, { Component } from 'react';
-import { ApiErrorBanner, DeleteModal, Loading, DisplayDate } from 'src/components';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
```
Comment (shrirupa) [inline src/pages/alerts/ListPage.js:2]: 🎉

**Review by shrirupa — APPROVED** [review]
Looks good to me! just had one comment of using useModal hook instead of the useReducer [here](https://github.com/SparkPost/2web2ui/pull/1882/files?file-filters%5B%5D=.js&file-filters%5B%5D=.scss&hide-deleted-files=true&file-filters%5B%5D=.snap&w=1#r522211489)

**Review by shrirupa — APPROVED** [review]

**Pull request merged**